src/providers/dart.rs

```diff
@@ -0,0 +1,58 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+use anyhow::{Context, Result};
+use serde::Deserialize;
+
+pub const DEFAULT_DART_PKG_NAME: &'static &str = &"dart";
+
+#[derive(Deserialize, Debug)]
+pub struct DartPubspec {
+    pub name: String,
+    pub version: String,
+}
+
+pub struct DartProvider {}
+
+impl Provider for DartProvider {
+    fn name(&self) -> &str {
+        "dart"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("pubspec.yaml"))
+    }
+
+    fn setup(&self, _app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        Ok(Some(SetupPhase::new(vec![Pkg::new(DEFAULT_DART_PKG_NAME)])))
+    }
+
+    fn install(&self, _app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
+        Ok(Some(InstallPhase::new("dart pub get".to_string())))
```

Comment:
This will allow better caching.
```suggestion
        let mut install_cmd = InstallPhase::new("dart pub get".to_string());
        install_cmd.add_file_dependency("pubspec.yaml".to_string()); 
        Ok(Some(install_cmd))
```

Reply:
Huh? How does this improve caching?

Reply:
If there are no changes in `pubspec.yaml` it'll use cached dependencies.